### ISSUE TITLE
require-direct-export rule doesn't work with `export default Vue.extend({})`

### ISSUE THREAD

**Issue by SimonSiefke** [vuejs/eslint-plugin-vue#907]
**Tell us about your environment**	
* **ESLint version:**	5.16.0
* **eslint-plugin-vue version:**	5.0.0
* **Node version:**	12.2.0

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: ['plugin:vue/essential', '@vue/prettier', '@vue/typescript'],
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'vue/require-direct-export': 'error',
  },
  parserOptions: {
    parser: '@typescript-eslint/parser',
  },
}

```

**What did you do?**	

```vue
<template>
  <div id="app">
    <img alt="Vue logo" src="./assets/logo.png" />
    <HelloWorld msg="Welcome to Your Vue.js + TypeScript App" />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import HelloWorld from "./components/HelloWorld.vue";

export default Vue.extend({
  name: "app",
  components: {
    HelloWorld
  }
});
</script>

<style>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
```

**What did you expect to happen?**
No error

**What actually happened?**
```
  12:1  error  Expected the component literal to be directly exported  vue/require-direct-export

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
```

**Comment by g-plane** [comment]
@ota-meshi @mysticatea @michalsnik This issue was opened one year ago and so far there's no response.

Is that behavior of `require-direct-export` by design?
